src/main/java/hlf/java/rest/client/service/impl/NetworkStatusImpl.java

```diff
@@ -55,6 +63,67 @@ private String getDeserializedConfig(MessageOrBuilder message) {
     return Base64.getEncoder().encodeToString(channelConfigString.getBytes());
   }
 
+  @Override
+  public ResponseEntity<ClientResponseModel> getAnchorPeerForChannel(String channelName) {
+    Network network = gateway.getNetwork(channelName);
+    Set<String> anchorPeersSet = new HashSet<>();
+    try {
+      if (network != null) {
+        Channel selectedChannel = network.getChannel();
+        byte[] selectedChannelBytes = selectedChannel.getChannelConfigurationBytes();
+        log.debug("peers: {} ", selectedChannel.getPeers());
+        Map<String, ConfigGroup> groupsMap =
+            Config.parseFrom(selectedChannelBytes).getChannelGroup().getGroupsMap();
+        ConfigGroup application =
+            groupsMap.get(FabricClientConstants.CHANNEL_CONFIG_GROUP_APPLICATION);
+        // Get all MSP IDs for a particular channel
+        Collection<String> peersOrganizationMSPIDs = selectedChannel.getPeersOrganizationMSPIDs();
+        log.debug("peersOrganizationMSPIDs: {}", peersOrganizationMSPIDs.toString());
+        for (String peerOrganizationMSPID : peersOrganizationMSPIDs) {
+          ConfigGroup peerOrgConfigGroup = application.getGroupsMap().get(peerOrganizationMSPID);
+          if (peerOrgConfigGroup != null) {
+            Map<String, Configtx.ConfigValue> valuesMap = peerOrgConfigGroup.getValuesMap();
+            Configtx.ConfigValue anchorPeers =
+                valuesMap.get(FabricClientConstants.CHANNEL_CONFIG_GROUP_VALUE_ANCHORPEERS);
+            if (null != anchorPeers && anchorPeers.getValue() != null) {
+              Configuration.AnchorPeers anchorPeersValue =
+                  Configuration.AnchorPeers.parseFrom(anchorPeers.getValue());
+              List<Configuration.AnchorPeer> anchorPeersList =
+                  anchorPeersValue.getAnchorPeersList();
+              if (anchorPeersList != null) {
+                for (Configuration.AnchorPeer anchorPeer : anchorPeersList) {
+                  // Concatenating the host and port to form a URL
+                  anchorPeersSet.add(anchorPeer.getHost() + ":" + anchorPeer.getPort());
+                }
+              }
+            }
+          }
+        }
+      }
+
+    } catch (InvalidArgumentException e) {
+      log.warn(
+          "Error while fetching channel config: Channel has no peer or orderer defined. Can not get configuration block");
+      throw new ServiceException(
+          ErrorCode.HYPERLEDGER_FABRIC_TRANSACTION_ERROR,
+          "Channel has no peer or orderer defined. Can not get configuration block",
+          e);
+    } catch (TransactionException e) {
+      log.warn("Error retrieving anchor peers: {} ", e.getMessage());
+      throw new FabricTransactionException(
+          ErrorCode.HYPERLEDGER_FABRIC_TRANSACTION_ERROR,
+          ErrorCode.HYPERLEDGER_FABRIC_TRANSACTION_ERROR.name(),
+          e);
```

Comment:
[nitpick] Can you kindly refactor this code? Aparently, we have nested loops and conditionals within a large try block. Please extract out methods wherever relevant.

Reply:
Thanks, Addressed.